src/main/java/net/minecraftforge/client/model/ModelLoaderRegistry2.java

```diff
@@ -0,0 +1,507 @@
+                }
+            }
+
+            return textures;
+        }
+
+        public static class Loader implements IModelLoader<VanillaProxy>
+        {
+            public static final Loader INSTANCE = new Loader();
+
+            private Loader()
+            {
+            }
+
+            @Override
+            public void onResourceManagerReload(IResourceManager resourceManager)
+            {
+
+            }
+
+            @Override
+            public VanillaProxy read(JsonDeserializationContext deserializationContext, JsonObject modelContents)
+            {
+                List<BlockPart> list = this.getModelElements(deserializationContext, modelContents);
+                return new VanillaProxy(list);
+            }
+
+            private List<BlockPart> getModelElements(JsonDeserializationContext deserializationContext, JsonObject object) {
+                List<BlockPart> list = Lists.newArrayList();
+                if (object.has("elements")) {
+                    for(JsonElement jsonelement : JSONUtils.getJsonArray(object, "elements")) {
+                        list.add(deserializationContext.deserialize(jsonelement, BlockPart.class));
+                    }
+                }
+
+                return list;
+            }
+        }
+    }
+
+    public static class ExpandedBlockModelDeserializer extends BlockModel.Deserializer
+    {
+        public static final Gson INSTANCE = (new GsonBuilder())
+                .registerTypeAdapter(BlockModel.class, new ExpandedBlockModelDeserializer())
+                .registerTypeAdapter(BlockPart.class, new BlockPart.Deserializer())
+                .registerTypeAdapter(BlockPartFace.class, new BlockPartFace.Deserializer())
+                .registerTypeAdapter(BlockFaceUV.class, new BlockFaceUV.Deserializer())
+                .registerTypeAdapter(ItemTransformVec3f.class, new ItemTransformVec3f.Deserializer())
+                .registerTypeAdapter(ItemCameraTransforms.class, new ItemCameraTransforms.Deserializer())
+                .registerTypeAdapter(ItemOverride.class, new ItemOverride.Deserializer())
+                .registerTypeAdapter(TRSRTransformation.class, ForgeBlockStateV1.TRSRDeserializer.INSTANCE)
+                .create();
+
+        public BlockModel deserialize(JsonElement element, Type targetType, JsonDeserializationContext deserializationContext) throws JsonParseException {
+            BlockModel model = super.deserialize(element, targetType, deserializationContext);
+            JsonObject jsonobject = element.getAsJsonObject();
+            IModelGeometry<?> geometry = deserializeGeometry(deserializationContext, jsonobject);
+
+            List<BlockPart> elements = model.getElements();
+            if (geometry != null) {
+                elements.clear();
+                model.customData.setCustomGeometry(geometry);
+            }
+
+            IModelState modelState = deserializeModelTransforms(deserializationContext, jsonobject);
```

Comment:
Instead of calling these manually, shouldn't these types get their own type adapters and then use `context.deserialize` here?

Reply:
This method does more than just read a TRSRTransformation object, it also handles things like the stock entries such as "forge:default-item". I might be able to change the logic in 1.15, but in this PR it doesn't make sense as this is a shared method with the old loader and I'd rather not do major refactoring on the old loader.